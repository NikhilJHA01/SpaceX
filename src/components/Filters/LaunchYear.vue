<template>
  <el-checkbox-group
    v-model="launchYearArray"
    size="small"
    text-color="#000000"
    fill="#5daf34"
    @change="changeFilter()"
  >
    <el-checkbox-button v-for="year in years" :label="year" :key="year">{{
      year
    }}</el-checkbox-button>
  </el-checkbox-group>
</template>

<script>
import { LaunchYears } from "../../utility/constant.js";

export default {
  name: "LaunchYear",
  data() {
    return {
      launchYearArray: [],
      years: LaunchYears
    };
  },
  methods: {
    // Emit event on change of filters
    changeFilter() {
      this.$store.dispatch("Set_Filter_By_LaunchYear", this.launchYearArray);
      this.$emit("changeFilter");
    }
  }
};
</script>
