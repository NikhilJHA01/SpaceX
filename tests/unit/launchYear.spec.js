import { mount } from "@vue/test-utils";
import LaunchYear from "@/components/Filters/LaunchYear";
import { LaunchYears } from "./data/flightTestData";

describe("Layout Years Component", () => {
  const wrapper = mount(LaunchYear, {
    data() {
      return {
        launchYearArray: [],
        years: LaunchYears
      };
    }
  });

  it("renders the Layout Year", async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a checkbox-group", () => {
    const checkbox_group = wrapper.find("el-checkbox-group");
    expect(checkbox_group.exists()).toBe(true);
  });

  it("Filter is not selected by default", () => {
    expect(wrapper.vm.$data.launchYearArray).toHaveLength(0);
  });
});
