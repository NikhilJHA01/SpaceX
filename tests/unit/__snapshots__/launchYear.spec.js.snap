// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Years Component renders the Layout Year 1`] = `
<el-checkbox-group size="small" text-color="#000000" fill="#5daf34">
  <el-checkbox-button label="2006">2006</el-checkbox-button>
  <el-checkbox-button label="2007">2007</el-checkbox-button>
  <el-checkbox-button label="2008">2008</el-checkbox-button>
  <el-checkbox-button label="2009">2009</el-checkbox-button>
  <el-checkbox-button label="2010">2010</el-checkbox-button>
  <el-checkbox-button label="2011">2011</el-checkbox-button>
  <el-checkbox-button label="2012">2012</el-checkbox-button>
  <el-checkbox-button label="2013">2013</el-checkbox-button>
  <el-checkbox-button label="2014">2014</el-checkbox-button>
  <el-checkbox-button label="2015">2015</el-checkbox-button>
  <el-checkbox-button label="2016">2016</el-checkbox-button>
  <el-checkbox-button label="2017">2017</el-checkbox-button>
  <el-checkbox-button label="2018">2018</el-checkbox-button>
  <el-checkbox-button label="2019">2019</el-checkbox-button>
  <el-checkbox-button label="2020">2020</el-checkbox-button>
</el-checkbox-group>
`;

exports[`Layout Years Component renders the Layout Year Card 1`] = `
<el-checkbox-group size="small" text-color="#000000" fill="#5daf34">
  <el-checkbox-button label="2006">2006</el-checkbox-button>
  <el-checkbox-button label="2007">2007</el-checkbox-button>
  <el-checkbox-button label="2008">2008</el-checkbox-button>
  <el-checkbox-button label="2009">2009</el-checkbox-button>
  <el-checkbox-button label="2010">2010</el-checkbox-button>
  <el-checkbox-button label="2011">2011</el-checkbox-button>
  <el-checkbox-button label="2012">2012</el-checkbox-button>
  <el-checkbox-button label="2013">2013</el-checkbox-button>
  <el-checkbox-button label="2014">2014</el-checkbox-button>
  <el-checkbox-button label="2015">2015</el-checkbox-button>
  <el-checkbox-button label="2016">2016</el-checkbox-button>
  <el-checkbox-button label="2017">2017</el-checkbox-button>
  <el-checkbox-button label="2018">2018</el-checkbox-button>
  <el-checkbox-button label="2019">2019</el-checkbox-button>
  <el-checkbox-button label="2020">2020</el-checkbox-button>
</el-checkbox-group>
`;
